src/Microsoft.Diagnostics.Monitoring.WebApi/EgressValidationAttribute.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Primitives;
+using Microsoft.Extensions.Logging;
+using System;
+
+namespace Microsoft.Diagnostics.Monitoring.WebApi
+{
+    [AttributeUsage(AttributeTargets.Method | AttributeTargets.Class)]
+    public sealed class EgressValidationAttribute : ActionFilterAttribute, IFilterFactory
+    {
+        public bool IsReusable => true;
+
+        IFilterMetadata IFilterFactory.CreateInstance(IServiceProvider serviceProvider)
+        {
+            var egressOutputOptions = serviceProvider.GetRequiredService<IEgressOutputOptions>();
+
+            EgressValidationActionFilter actionFilter = new EgressValidationActionFilter(egressOutputOptions);
+
+            return actionFilter;
+        }
+    }
+
+    [AttributeUsage(AttributeTargets.Class)]
+    public class EgressValidationUnhandledExceptionFilter : ActionFilterAttribute, IExceptionFilter
+    {
+        private readonly ILogger _logger;
+
+        public EgressValidationUnhandledExceptionFilter(ILogger<EgressValidationUnhandledExceptionFilter> logger)
+        {
+            _logger = logger;
+        }
+
+        public void OnException(ExceptionContext context)
+        {
+            if (context.Exception is EgressValidationActionFilter.EgressDisabledException egressException)
+            {
+                BadRequestObjectResult badRequestResult = new BadRequestObjectResult(egressException.Message);
+                badRequestResult.Value = ExceptionExtensions.ToProblemDetails(egressException, StatusCodes.Status400BadRequest);
+
+                context.Result = badRequestResult;
```

Comment:
Can you check that the HTTP response (using PostMan, curl, etc) report the correct content-type (application/problem+json)? If it doesn't, you might have to add `result.ContentTypes.Add(ContentTypes.ApplicationProblemJson);` to the result before returning.

Reply:
Can confirm; Postman says "application/problem+json; charset=utf-8"